android-components/components/browser/menu2/src/main/java/mozilla/components/browser/menu2/ext/PopupWindow.kt

```diff
@@ -29,226 +20,25 @@ internal fun PopupWindow.displayPopup(
     containerView: View,
     anchor: View,
     orientation: Orientation? = null,
+    style: MenuStyle? = null,
 ): Boolean {
-    // Measure menu.
-    val spec = View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED)
-    containerView.measure(spec, spec)
-
-    val containerRecyclerView =
-        containerView.findViewById<RecyclerView>(R.id.mozac_browser_menu_recyclerView)
-    val recyclerViewAdapter = containerRecyclerView.adapter
-
-    if (recyclerViewAdapter == null) {
-        Log.log(
-            priority = Log.Priority.ERROR,
-            tag = "mozac-popupwindow",
-            message = "The browser menu recyclerview has no adapter set.",
-        )
-        return false
-    }
-
-    val containerViewHeight = containerView.measuredHeight
-    val recyclerViewHeight = containerRecyclerView.measuredHeight
-    val recyclerViewItemCount = recyclerViewAdapter.itemCount
+    val positioningData =
+        inferMenuPositioningData(containerView, anchor, style, orientation) ?: return false
 
-    val hasViewInitializedCorrectly =
-        containerViewHeight > 0 && recyclerViewHeight > 0 && recyclerViewItemCount > 0
-    if (!hasViewInitializedCorrectly) {
-        Log.log(
-            priority = Log.Priority.ERROR,
-            tag = "mozac-popupwindow",
-            message = "containerViewHeight: $containerViewHeight must be more than 0," +
-                "recyclerViewHeight: $recyclerViewHeight must be more than 0 &" +
-                "recyclerViewItemCount: $recyclerViewItemCount must be more than 0.",
-        )
-        return false
-    }
-
-    // Popup window does not need a input method. This avoids keyboard flicker when menu is opened.
     inputMethodMode = PopupWindow.INPUT_METHOD_NOT_NEEDED
 
-    showPopup(orientation, anchor, recyclerViewHeight, recyclerViewItemCount, containerViewHeight)
+    showAtAnchorLocation(anchor, positioningData)
```

Comment:
The reason behind switching to `showAtAnchorLocation` is the manual setting of [height](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/browser/menu2/src/main/java/mozilla/components/browser/menu2/ext/PopupWindow.kt#197) for the popup. It makes the [PopupWindow](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/widget/PopupWindow.java#1859) to move the menu up offscreen in there is keyboard opened, ignoring any provided offsets, making precise positioning impossible.